diff -aruN loci/dockerfiles/centos8_stream/CentOS.repo loci/dockerfiles/centos8_stream/CentOS.repo
--- loci/dockerfiles/centos8_stream/CentOS.repo	1970-01-01 09:00:00.000000000 +0900
+++ loci/dockerfiles/centos8_stream/CentOS.repo	2023-11-17 14:06:44.492425802 +0900
@@ -0,0 +1,46 @@
+[appstream]
+name=CentOS Linux $releasever - AppStream
+baseurl=http://%%PACKAGE_MIRROR%%/$contentdir/$stream/AppStream/$basearch/os/
+gpgcheck=1
+gpgkey=file:///etc/pki/rpm-gpg/RPM-GPG-KEY-centosofficial
+
+[baseos]
+name=CentOS Linux $releasever - BaseOS
+baseurl=http://%%PACKAGE_MIRROR%%/$contentdir/$stream/BaseOS/$basearch/os/
+gpgcheck=1
+gpgkey=file:///etc/pki/rpm-gpg/RPM-GPG-KEY-centosofficial
+
+[extras]
+name=CentOS Linux $releasever - Extras
+baseurl=http://%%PACKAGE_MIRROR%%/$contentdir/$stream/extras/$basearch/os/
+gpgcheck=1
+gpgkey=file:///etc/pki/rpm-gpg/RPM-GPG-KEY-centosofficial
+
+[powertools]
+name=CentOS Linux $releasever - PowerTools
+baseurl=http://%%PACKAGE_MIRROR%%/$contentdir/$stream/PowerTools/$basearch/os/
+gpgcheck=1
+gpgkey=file:///etc/pki/rpm-gpg/RPM-GPG-KEY-centosofficial
+
+#[openstack]
+#name=CentOS-$releasever - OpenStack
+#baseurl=http://%%PACKAGE_MIRROR%%/centos/$stream/cloud/$basearch/openstack-yoga/
+#gpgcheck=1
+#gpgkey=file:///etc/pki/rpm-gpg/RPM-GPG-KEY-CentOS-SIG-Cloud
+
+[ceph]
+name=Ceph packages for $basearch
+baseurl=https://download.ceph.com/rpm-quincy/el8/$basearch
+enabled=1
+priority=2
+gpgcheck=1
+gpgkey=https://download.ceph.com/keys/release.asc
+
+[ceph-noarch]
+name=Ceph noarch packages
+baseurl=https://download.ceph.com/rpm-quincy/el8/noarch
+enabled=1
+priority=2
+gpgcheck=1
+gpgkey=https://download.ceph.com/keys/release.asc
+
diff -aruN loci/dockerfiles/centos8_stream/Dockerfile patches/loci/dockerfiles/centos8_stream/Dockerfile
--- loci/dockerfiles/centos8_stream/Dockerfile	1970-01-01 09:00:00.000000000 +0900
+++ loci/dockerfiles/centos8_stream/Dockerfile	2023-11-17 16:25:01.748412738 +0900
@@ -0,0 +1,16 @@
+ARG FROM=quay.io/centos/centos:stream8
+FROM ${FROM}
+
+ARG PACKAGE_MIRROR=mirror.centos.org
+ARG PIP_INDEX_URL=https://pypi.python.org/simple/
+ARG PIP_TRUSTED_HOST=pypi.python.org
+ENV PIP_INDEX_URL=${PIP_INDEX_URL}
+ENV PIP_TRUSTED_HOST=${PIP_TRUSTED_HOST}
+
+RUN rm -rf /etc/yum.repos.d/*
+COPY CentOS.repo /etc/yum.repos.d/
+#COPY RPM-GPG-KEY-CentOS-SIG-Cloud /etc/pki/rpm-gpg/
+RUN sed -i "s|%%PACKAGE_MIRROR%%|${PACKAGE_MIRROR}|g" /etc/yum.repos.d/CentOS.repo \
+  && dnf update -y
+# NOTE(aostapenko) repositories are restored on yum update, deleting them again
+RUN find /etc/yum.repos.d/ -type f ! -name 'CentOS.repo' -delete
diff -aruN loci/dockerfiles/centos8_stream/RPM-GPG-KEY-CentOS-SIG-Cloud patches/loci/dockerfiles/centos8_stream/RPM-GPG-KEY-CentOS-SIG-Cloud
--- loci/dockerfiles/centos8_stream/RPM-GPG-KEY-CentOS-SIG-Cloud	1970-01-01 09:00:00.000000000 +0900
+++ loci/dockerfiles/centos8_stream/RPM-GPG-KEY-CentOS-SIG-Cloud	2023-11-01 18:59:14.708471186 +0900
@@ -0,0 +1,20 @@
+-----BEGIN PGP PUBLIC KEY BLOCK-----
+Version: GnuPG v2.0.22 (GNU/Linux)
+
+mQENBFVWcCcBCACfm3eQ0526/I0/p7HpR0NjK7K307XHhnbcbZv1sDUjQABDaqh0
+N4gnZcovf+3fj6pcdOmeOpGI0cKE7Fh68RbEIqyjB7l7+j1grjewR0oCFFZ38KGm
+j+DWQrj1IJW7JU5fH/G0Cu66ix+dJPcuTB3PJTqXN3ce+4TuG09D+epgwfbHlqaT
+pH2qHCu2uiGj/AaRSM/ZZzcInMaeleHSB+NChvaQ0W/m+kK5d/20d7sfkaTfI/pY
+SrodCfVTYxfKAd0TLW03kimHs5/Rdz+iZWecVKv6aFxzaywbrOjmOsy2q0kEWIwX
+MTZrq6cBRRuWyiXsI2zT2YHQ4UK44IxINiaJABEBAAG0WkNlbnRPUyBDbG91ZCBT
+SUcgKGh0dHA6Ly93aWtpLmNlbnRvcy5vcmcvU3BlY2lhbEludGVyZXN0R3JvdXAv
+Q2xvdWQpIDxzZWN1cml0eUBjZW50b3Mub3JnPokBOQQTAQIAIwUCVVZwJwIbAwcL
+CQgHAwIBBhUIAgkKCwQWAgMBAh4BAheAAAoJEPm5/ud2RCnmATUH/3HDtWxpFkmy
+FiA3VGkMt5dp3bgCRSd84X6Orfx1LARowpI4LomCGglGBGXVJePBacwcclorbLaz
+uWrW/wU0efz0aDB5c4NPg/yXfNvujvlda8ADJwZXVBQphzvaIKwl4PqBsEnxC10I
+93T/0iyphAhfMRJ5R8AbEHMj7uF+TWTX/JoyQagllMqWTwoP4DFRutPdOmmjwvSV
+kWItH7hq6z9+M4dhlqeoOvPbL5oCxX7TVmLck02Q5gI4syULOa7sqntzUQKFkhWp
+9U0+5KrBQBKezrurrrkq/WZR3WNE1KQfNQ77f7S2JcXJdOaKgJ7xe7Y2flPq98Aq
+wKXK7l1c3dc=
+=W6yF
+-----END PGP PUBLIC KEY BLOCK-----
