--- openstack-helm-images/mariadb/build.sh	2025-01-17 16:59:06.462510403 +0900
+++ patches/openstack-helm-images/mariadb/build.sh	2025-01-17 16:58:12.962540641 +0900
@@ -1,6 +1,10 @@
-#!/bin/bash
+#!/bin/bash -x
 SCRIPT=`realpath $0`
 SCRIPT_DIR=`dirname ${SCRIPT}`
+ENVFILE="${SCRIPT_DIR}/../../.env.mariadb"
+if [[ -f "${ENVFILE}" ]]; then
+  . ${ENVFILE}
+fi
 ## Only build from main folder
 cd ${SCRIPT_DIR}/..
 
