--- openstack-helm-images/mariadb/build.sh	2023-12-19 11:30:09.035612488 +0900
+++ patches/openstack-helm-images/mariadb/build.sh	2023-12-20 11:49:18.549886766 +0900
@@ -1,6 +1,10 @@
-#!/bin/bash
+#!/bin/bash -x
 SCRIPT=`realpath $0`
 SCRIPT_DIR=`dirname ${SCRIPT}`
+## get .env in SCRIPT_DIR
+if [[ -f "${SCRIPT_DIR}/../../.env" ]]; then
+  . .env
+fi
 ## Only build from main folder
 cd ${SCRIPT_DIR}/..
 
